File: python/strategy/DuckSimulator/client/RubberDuck.py
from client.Duck import Duck
from behavior.QuackBehavior import MuteQuack
from behavior.FlyBehavior import FlyNoWay

class RubberDuck(Duck):

    def __init__(self, duck_name):
        super().__init__(duck_name)
        self._fly_behavior = FlyNoWay(duck_name)
        self._quack_behavior = MuteQuack(duck_name)

File: ruby/strategy/DuckSimulator/client/rubber_duck.rb
require "./client/duck"
require './behavior/fly_behavior'
require './behavior/quack_behavior'

class RubberDuck < Duck

  def initialize(duck_name)
    super(duck_name)
    @fly_behavior = FlyNoWay.new(duck_name)
    @quack_behavior = MuteQuack.new(duck_name)
  end

end